fix: Various fixes and updates for Fether FAQ relating to PR #317 and Fether PR #471 (#312)
paritytech/wiki: Fether-FAQ.md

### Why does the sync starts back at 0% when I restart Fether?

The background process to synchronize the blockchain does not restart from scratch. The sync continues from the point it was left at the last time Fether was launched.

### How to delete an account?

We chose not to allow the deletion of accounts from the Fether user interface. In case you really know what you are doing and do not want an account to appear in Fether, you can locate and move the keystore file from the 'keys' folder. You will find it here:

- macOS: `~/Library/Application\ Support/io.parity.ethereum/keys/ethereum/`
- Linux: `$HOME/.local/share/io.parity.ethereum/keys`
- Windows 7/10: `%HOMEPATH%/AppData/Roaming/Parity/Ethereum/keys`

### Where is Fether's production binary executable file installed?

> **TODO**

Reviewer: Windows:
  - Parity Fether `C:\Users\fether\AppData\Local\Programs\fether\fether`
  - Parity Ethereum `C:\Users\fether\AppData\Local\Programs\fether\resources\static\parity`
  - Parity Ethereum chain `C:\Users\fether\AppData\Roaming\Parity\Ethereum\`

Linux:
  - Parity Fether (for `.deb`) `/opt/Parity Fether/fether`
  - Parity Ethereum `/opt/Parity Fether/resources/static/parity`
  - Parity Ethereum chain `~/.local/share/io.parity.ethereum/`

Author: It mentions some different paths here for Linux and Windows: https://github.com/paritytech/js-libs/blob/master/packages/electron/src/getParityPath.ts#L63

### Where is the Parity Ethereum binary executable file installed?

- macOS: `/Applications/Parity Ethereum.app/Contents/MacOS/parity`
- Linux: `/bin/parity', '/usr/bin/parity', or '/usr/local/bin/parity`

> - Windows: `C:\Program Files\Parity Technologies\Parity\parity.exe`

Reviewer: These have just changed with the latest PR, see my previous comment.

Author: It looks like this library needs to be updated then. What do you think?
https://github.com/paritytech/js-libs/blob/master/packages/electron/src/getParityPath.ts#L63

Reviewer: yes

### How to launch Fether on a different network?

You can pass specific flags (such as `--chain`) for Fether to launch the underlying Parity Ethereum on a specific network:

```bash
# Launching Fether with a Parity Ethereum light client node on Ropsten
$ /path/to/fether --chain ropsten --light
```

### How to launch Fether with a separately launched Parity Ethereum node

Fether will download and install Parity Ethereum if it is not already installed. Fether will run a Parity Ethereum light client node if Parity Ethereum has not already been launched separately.

1) Launch a Parity Ethereum node using any appropriate flags. Check what flags are available with `parity --help`).

```bash
# Launch a Parity Ethereum light client node on the Ropsten network instead of the Kovan (default) test network
$ parity --chain ropsten --light
```

> 2) Launch Fether (using the GUI or in a terminal session), it will automatically detect the local node and connect to it.

Reviewer: "GUI" might be a little too technical, can we talk about" launcher menu" or anything that you see fit?

Reviewer: There's still GUI here :)
```suggestion
2) Launch Fether (using the launcher menu or in a terminal session), it will automatically detect the local node and connect to it.
```

> ### How to connect Fether to a full Parity Ethereum node?

Reviewer: Parity Ethereum full node?

Fether will automatically connect to any Parity Ethereum node that is already running (using WebSockets port 8546).
To connect Fether to a full node:

1) Launch a Parity Ethereum full node using any appropriate flags. Check what flags are available with `parity --help`).

```bash
# Launch a Parity Ethereum full node on Ropsten test network
$ parity --chain ropsten
```

2) Launch Fether (using the GUI or in a terminal session), it will automatically detect the local node and connect to it.

### How to access logs to troubleshoot errors

To help the developers understand what is going on if Fether crashes, it is essential to provide logs.

> Fether is an [Electron](https://electronjs.org/) app with a Node.js backend (fether-electron package) and a React.js frontend (fether-react and fether-ui packages), where logs are provided using JavaScript's `console.log` or the Pino library.

Reviewer: We should be concise and talk to non devs here. No need to talk about electron, react, node, front/back-end as this doesn't answer the question about accessing logs.

Author: Ok, I just thought an initial context as to why there are frontend and backend logs would help users who are new to development or non devs and may be curious to explore the codebase.

1) Launch a Parity Ethereum node using any appropriate flags. Check what flags are available with `parity --help`).

```bash
# Launch a Parity Ethereum light client node on the Ropsten network instead of the Kovan (default) test network
$ parity --chain ropsten --light
```

2) Launch Fether (using the GUI or in a terminal session), it will automatically detect the local node and connect to it.

### How to connect Fether to a Parity Ethereum full node?

Fether will automatically connect to any Parity Ethereum node that is already running (using WebSockets port 8546).
To connect Fether to a full node:

1) Launch a Parity Ethereum full node using any appropriate flags. Check what flags are available with `parity --help`).

```bash
# Launch a Parity Ethereum full node on Ropsten test network
$ parity --chain ropsten
```

2) Launch Fether (from the application launcher menu or in a terminal session), it will automatically detect the local node and connect to it.

### How to access logs to troubleshoot errors

To help the users understand what is going on if Fether crashes, it is essential to provide logs.

#### Development Environment

> Frontend logs may be viewed as follows:

Reviewer: There really is no need to use tech jargon.
```suggestion
Logs for the user interface may be viewed as follows:
```

Author: Thanks, I've added `(UI)` too

- Open the Fether menu by right-clicking the Fether window
- Click the menu item "View > Toggle Developer Tools"
- Undock the Developer Tools into a separate window
  - Clicking the '3 dots' icon located at the top right
  - Select the 'separate windows' symbol
- Click the 'Console' tab in the Developer Tools window
- Expand the 'Console' tab so it is large enough to take a screenshot of any logs, warnings, or errors.
- Please take a screenshot of the Fether window and the logs and share them with us by posting an [Issue](https://github.com/paritytech/fether/issues) in the Fether GitHub repository or by clicking the 'Feedback' button on the Accounts page of the Fether window.

> If you want the Developer Tools to automatically open each time you run Fether, see the FAQ "How to open Developer Tools automatically when Fether launches".

Reviewer: Deep link?
```suggestion
If you want the Developer Tools to automatically open each time you run Fether, see [How to open Developer Tools automatically when Fether launches](#how-to-open-developer-tools-automatically-when-fether-launches).
```

Author: Good idea!

> Backend logs will appear in the terminal session window where you launch Fether (i.e. with `yarn electron` or `yarn start`).

Reviewer: Most user can launch Fether in the terminal without having the sources, hence I don't think we should talk about `yarn` commands here.

```suggestion
Logs related to the underlying Ethereum node or the overall Fether application will only appear if you launch Fether user a terminal.
```

Author: I've added a `Tips:` and `Advanced:` section, so the answer satisfies general users upfront without the technical jargon

> #### Production Environment

Reviewer: I don't think  it's needed to dissociate prod from dev environment since accessing the logs is the same.

### How to run Fether without "taskbar mode"

#### Development Environment

Disable the `TASKBAR` environment variable prior to launching Fether as follows:

```bash
TASKBAR=false yarn start
```

### How to open Developer Tools automatically when Fether launches

#### Development Environment

Open Developer Tools automatically by running Fether with the `DEBUG` environment variable set as follows:

```bash
DEBUG=true yarn start
```

### What to do if the Fether window appears unresponsive or does not load

#### Development Environment

If you run Fether with `yarn start` it is a known issue that the Fether window will hang with just a white/blank screen even after it has compiled successfully. Fix it by simply right-clicking the Fether window to reveal the Fether menu, and choose the menu item 'View > Reload' (or use shortcut CMD + R on macOS, CTRL + R on Linux/Windows). Alternatively run Fether with `yarn electron` (without live reload).

If Fether is otherwise unresponsive or does not load then please follow the steps in the FAQ "How to access logs to troubleshoot errors".

#### Production Environment

> Please follow the steps in the FAQ "How to access logs to troubleshoot errors".

Reviewer: Deep link?

```suggestion
Please follow the steps in the FAQ [How to access logs to troubleshoot errors](#how-to-access-logs-to-troubleshoot-errors).
```

- Linux: `$HOME/.local/share/io.parity.ethereum/keys`
- Windows 7/10: `%HOMEPATH%/AppData/Roaming/Parity/Ethereum/keys`

### Where is Fether installed?

- macOS: `/Applications/Parity Ethereum.app/Contents/MacOS/parity`
- Linux:
  - `.deb`
    - Parity Fether: `/opt/Parity\ Fether/fether`
  - AppImage
    - Parity Fether: `/wherever_you_store_it/Parity\ Fether\ X.X.X.AppImage`
  - binary
    - Parity Fether: `/wherever_you_store_it/fether`
- Windows:
  - Parity Fether: `C:\Users\fether\AppData\Local\Programs\fether\fether`

### Where is Parity Ethereum installed?

- macOS: `/Applications/Parity Ethereum.app/Contents/MacOS/parity`
- Linux:
  - `.deb`
    - Parity Ethereum: `/opt/Parity Fether/resources/static/parity`
    - Parity Ethereum chain: `~/.local/share/io.parity.ethereum/`
  - AppImage
    - Parity Ethereum: `/tmp/.mount_Parity[something]/resources/static/parity`
    - Parity Ethereum chain: `~/.local/share/io.parity.ethereum/`
  - binary

>     - Parity Ethereum: `/bin/parity`, `/usr/bin/parity`, or `/usr/local/bin/parity`

Reviewer: These were either installed separately, or with older versions of Fether. I think we should  focus on `/resources/static/parity` and eventually add at the end something like "parity can also be found here if installed independantely or with Fether <0.3"
